DFC.ServiceTaxonomy.CompUi/Repository/DataRepository.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Data.Common;
+using DFC.ServiceTaxonomy.CompUi.Dapper;
+using DFC.ServiceTaxonomy.CompUi.Interfaces;
+using DFC.ServiceTaxonomy.CompUi.Models;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data;
+
+namespace DFC.ServiceTaxonomy.CompUi.Repository
+{
+    public class DataService : IDataService
+    {
+        private readonly IDbConnectionAccessor _dbaAccessor;
+        private readonly ILogger<DataService> _logger;
+        private readonly IDapperWrapper _dapperWrapper;
+
+        public DataService(
+            IDbConnectionAccessor dbaAccessor,
+            IDapperWrapper dapperWrapper,
+            ILogger<DataService> logger)
+        {
+            _dapperWrapper = dapperWrapper;
+            _dbaAccessor = dbaAccessor;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<RelatedContentData>> GetRelatedContentDataByContentItemIdAndPage(Processing processing)
+        {
+            IEnumerable<RelatedContentData>? resultList = [];
+
+            var sql = $"SELECT DocumentId FROM RelatedContentItemIndex WITH (NOLOCK) WHERE RelatedContentIds LIKE '%{processing.ContentItemId}%' AND ContentType = 'Page' ";
+            var documentIds = await ExecuteQuery<string>(sql);
+
+            if (documentIds != null && documentIds.Count() > 0)
+            {
+                var queryIds = string.Join(",", documentIds);
+
+                sql = $"SELECT ContentItemId, DisplayText, Author, ContentType, " +
+                    $"JSON_VALUE(Content,'$.GraphSyncPart.Text') AS GraphSyncId, " +
+                    $"JSON_VALUE(Content,'$.PageLocationPart.FullUrl') AS FullPageUrl " +
+                    $"FROM Document AS D " +
+                    $"INNER JOIN ContentItemIndex AS CII on CII.DocumentId = D.Id " +
+                    $"WHERE D.Id IN ({queryIds}) " +
+                    $"AND Latest = 1 AND Published = 1";
+
+                resultList = await ExecuteQuery<RelatedContentData>(sql);
+            }
+
+            return resultList;
+        }
```

Comment:
This is really scary. I really hope you know where the data is coming from, as otherwise this seems like an easy target for SQL injection.

A string should never be put directly into a query, but parameterization should be used instead. Unfortunately, the wrapper that has been created does not seem to support this. I would highly encourage to get the wrapper updated to support parameterization and get this query in better shape!

https://www.learndapper.com/parameters